hibernate-core/src/test/java/org/hibernate/orm/test/fetch/GenerateSubselectsOnJoinedAssociationTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.fetch;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+
+import org.hibernate.Hibernate;
+import org.hibernate.annotations.Fetch;
+import org.hibernate.annotations.FetchMode;
+import org.hibernate.engine.spi.BatchFetchQueue;
+import org.hibernate.engine.spi.EntityKey;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.persister.entity.EntityPersister;
+
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+@Jpa(annotatedClasses = {
+		GenerateSubselectsOnJoinedAssociationTest.LazySelectRoot.class,
+		GenerateSubselectsOnJoinedAssociationTest.LazySelectNode.class
+})
```

Comment:
Small note, but if you want you can use `@DomainModel` instead of `@Jpa`, together with the `@SessionFactory` annotation they will allow your test methods to accept a `SessionFactoryScope` so you won't need to call `unwrap`.